metadata-ingestion/src/datahub/ingestion/source/data_lake.py

```diff
@@ -0,0 +1,991 @@
+    column_profile: DatasetFieldProfileClass
+
+    # if the histogram is a list of value frequencies (discrete data) or bins (continuous data)
+    histogram_distinct: Optional[bool] = None
+
+    type_: SparkDataType = NullType
+
+    unique_count: Optional[int] = None
+    non_null_count: Optional[int] = None
+    cardinality: Optional[Cardinality] = None
+
+
+class DataLakeProfilerConfig(ConfigModel):
+    enabled: bool = False
+
+    # These settings will override the ones below.
+    profile_table_level_only: bool = False
+
+    allow_deny_patterns: AllowDenyPattern = AllowDenyPattern.allow_all()
+
+    turn_off_expensive_profiling_metrics: bool = False
+
+    max_number_of_fields_to_profile: Optional[pydantic.PositiveInt] = None
+
+    include_field_null_count: bool = True
+    include_field_min_value: bool = True
+    include_field_max_value: bool = True
+    include_field_mean_value: bool = True
+    include_field_median_value: bool = True
+    include_field_stddev_value: bool = True
+    include_field_quantiles: bool = True
+    include_field_distinct_value_frequencies: bool = True
+    include_field_histogram: bool = True
+    include_field_sample_values: bool = True
+
+    @pydantic.root_validator()
+    def ensure_field_level_settings_are_normalized(
+        cls: "DataLakeProfilerConfig", values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        max_num_fields_to_profile_key = "max_number_of_fields_to_profile"
+        table_level_profiling_only_key = "profile_table_level_only"
+        max_num_fields_to_profile = values.get(max_num_fields_to_profile_key)
+        if values.get(table_level_profiling_only_key):
+            all_field_level_metrics: List[str] = [
+                "include_field_null_count",
+                "include_field_min_value",
+                "include_field_max_value",
+                "include_field_mean_value",
+                "include_field_median_value",
+                "include_field_stddev_value",
+                "include_field_quantiles",
+                "include_field_distinct_value_frequencies",
+                "include_field_histogram",
+                "include_field_sample_values",
+            ]
+            # Suppress all field-level metrics
+            for field_level_metric in all_field_level_metrics:
+                values[field_level_metric] = False
+            assert (
+                max_num_fields_to_profile is None
+            ), f"{max_num_fields_to_profile_key} should be set to None"
+
+        if values.get("turn_off_expensive_profiling_metrics"):
+            if not values.get(table_level_profiling_only_key):
+                expensive_field_level_metrics: List[str] = [
+                    "include_field_quantiles",
+                    "include_field_distinct_value_frequencies",
+                    "include_field_histogram",
+                    "include_field_sample_values",
+                ]
+                for expensive_field_metric in expensive_field_level_metrics:
```

Comment:
I'm not a huge fan of the code duplication 

We also probably don't need the `expensive_field_level_metrics` option if we change the defaults